runtime/src/accounts_index.rs

```diff
@@ -256,10 +276,19 @@ impl<'a, T: 'a + Clone> AccountsIndex<T> {
         self.previous_uncleaned_roots.remove(&slot);
     }
 
-    pub fn reset_uncleaned_roots(&mut self) -> HashSet<Slot> {
-        let empty = HashSet::new();
-        let new_previous = std::mem::replace(&mut self.uncleaned_roots, empty);
-        std::mem::replace(&mut self.previous_uncleaned_roots, new_previous)
+    pub fn reset_uncleaned_roots(&mut self, max_cleaned_slot: Option<Slot>) -> HashSet<Slot> {
```

Comment:
nitx: `max_clean_root` (to align with other sibling methods)

Reply:
agreed!